diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/ASTNodeUtilsTest.kt

```diff
@@ -62,51 +73,516 @@ class ASTNodeUtilsTest {
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        var firstCounter = 0
+        var secondCounter = 0
+        applyToCode(code){node ->
+            node.getAllChildrenWithType(CLASS).ifNotEmpty {
+                Assert.assertEquals(map { it.text }, listOf(code))
+                firstCounter++
+            }
+            if (node.getAllChildrenWithType(IDENTIFIER).isNotEmpty() && node.treeParent.elementType == FILE) {
+                Assert.assertEquals(node.getAllChildrenWithType(IDENTIFIER)[0].text, "Test")
+                secondCounter++
+            }
+        }
+        Assert.assertEquals(1, firstCounter)
+        Assert.assertEquals(1, secondCounter)
+    }
+
+    @Test
+    fun `test getFirstChildWithType`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        var counter = 0
+        applyToCode(code){node ->
+            if (node.getAllChildrenWithType(IDENTIFIER).isNotEmpty() && node.treeParent.elementType == FILE) {
+                Assert.assertEquals(node.getFirstChildWithType(IDENTIFIER)!!.text, "Test")
+                counter++
+            }
+        }
+        Assert.assertEquals(1, counter)
+    }
+
+    @Test
+    fun `test hasChildOfType`(){
+        val code = """
+            class Test {
+                val x = 0
+            }
+        """.trimIndent()
+        var counter = 0
+        applyToCode(code){node ->
+            if (node.getIdentifierName() != null) {
+                Assert.assertTrue(node.hasChildOfType(IDENTIFIER))
+                counter++
+            }
+        }
+        Assert.assertEquals(2, counter)
+    }
+
+    @Test
+    fun `test hasAnyChildOfTypes`(){
+        val code = """
+            class Test {
+                val x = 0
+            }
+        """.trimIndent()
+        var counter = 0
+        applyToCode(code){node ->
+            if (node.getAllChildrenWithType(IDENTIFIER).isNotEmpty() || node.getAllChildrenWithType(CLASS).isNotEmpty()) {
+                Assert.assertTrue(node.hasAnyChildOfTypes(IDENTIFIER, CLASS))
+                counter++
+            }
+        }
+        Assert.assertEquals(3, counter)
+    }
+
+    @Test
+    fun `test findChildBefore`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        var counter = 0
+        applyToCode(code){node ->
+            if (node.findChildBefore(CLASS_BODY, CLASS) != null) {
```

Comment:
I believe `CLASS_BODY` is child of `CLASS`, not sibling? If you dig into implementation of `findChildBefore`, you can see that if `beforeThisNodeType` is not found, `childNodeType` is simply searched among this node's children. Could you please check if these nodes are actually siblings and add one more example for another case?